bus-mapping/src/evm/opcodes/mload.rs

```diff
@@ -97,81 +97,71 @@ impl Opcode for Mload {
 mod mload_tests {
     use super::*;
     use crate::{
+        bytecode,
         evm::{
-            EvmWord, GasCost, GasInfo, Memory, OpcodeId, ProgramCounter, Stack,
+            EvmWord, GasCost, GasInfo, OpcodeId, ProgramCounter, Stack,
             StackAddress,
         },
-        BlockConstants, ExecutionTrace,
+        external_tracer, BlockConstants, ExecutionTrace,
     };
     use pasta_curves::pallas::Scalar;
 
+    macro_rules! gas_info {
+        ($gas:ident, $gas_cost: ident) => {{
+            #[allow(unused_assignments)]
+            GasInfo {
+                gas: {
+                    let temp = $gas;
+                    $gas -= GasCost::$gas_cost.as_usize() as u64;
+                    temp
+                },
+                gas_cost: GasCost::$gas_cost,
+            }
+        }};
+    }
+
     #[test]
     fn mload_opcode_impl() -> Result<(), Error> {
-        let trace = r#"
-        [
-            {
-            "pc": 7,
-            "op": "MLOAD",
-            "gas": 79,
-            "gasCost": 3,
-            "depth": 1,
-            "stack": [
-                "40"
-            ],
-            "memory": [
-                "0000000000000000000000000000000000000000000000000000000000000000",
-                "0000000000000000000000000000000000000000000000000000000000000000",
-                "0000000000000000000000000000000000000000000000000000000000000080"
-            ]
-            },
-              {
-                "pc": 8,
-                "op": "STOP",
-                "gas": 76,
-                "gasCost": 0,
-                "depth": 1,
-                "stack": [
-                  "80"
-                ],
-                "memory": [
-                  "0000000000000000000000000000000000000000000000000000000000000000",
-                  "0000000000000000000000000000000000000000000000000000000000000000",
-                  "0000000000000000000000000000000000000000000000000000000000000080"
-                ]
-              }
-        ]
-        "#;
+        let code = bytecode! {
+            // Setup state
+            PUSH1(0x80u64)
+            PUSH1(0x40u64)
+            MSTORE
+
+            PUSH1(0x40u64)
+            // Start byte code tested
+            [start]
+            MLOAD
+        };
+
+        let block_ctants = BlockConstants::default();
+
+        // Get the execution steps from the external tracer
+        let obtained_steps = &external_tracer::trace(&block_ctants, &code)
+            [code.get_pos("start")..];
```

Comment:
I would probably add some method inside `ByteCode` like: `code_from_pos()` or similar. So that we abstract over the Indexing op. 

Just a suggestion anyway :)

Reply:
Here all the code needs to be sent to geth because the code we actually want to test also depends on the memory modifications before it so geth needs to execute as well. So I don't think we can do this range on the `Bytecode` level, it needs to be done on the execution steps. And I didn't want to create dependencies between the two objects so I left it like this. :)

Reply:
Ohhh I see. 
Makes sense now! Apologies for the missundestanding :)